### DISCUSSION TITLE
How to access Pimcore backend profile Pictures?

### DISCUSSION THREAD

**Discussion by idna001** [pimcore/pimcore#16205] [Q&A / Support] [answered]
Hello, 
I would like to access the profile Picture of the Pimcore Backend User in frontend.
My object used the Data Component 'User' in my class Object.

I can access the things like Name and ID, but not the URL of picture. 
The only URL I'm getting is the thumbnail Image, but this only works while logged in into the Pimcore backend:
`/admin/user/get-image?id=2`

Here is my current code: 
```php
$user = User::getById($request);
$name = $user->getFullName();
$avatar = $user->getImage();

``` 

This is giving me the stream resource: 
```php
"userAdmin" => array:2 [▼
    "name" => "Pimcore Admin"
    "avatar" => stream resource @24 ▼
      timed_out: false
      blocked: true
      eof: false
      wrapper_type: "plainfile"
      stream_type: "STDIO"
      mode: "rb"
      unread_bytes: 0
      seekable: true
      uri: "[/home/admin/www/dev-overview/var/admin/user-image/user-thumbnail-2.png](file:///home/admin/www/dev-overview/var/admin/user-image/user-thumbnail-2.png#L0)"
      options: []
    }
  ]
``` 

May I can use the stream resource to access this outside of the Admin?

Thanks for any tips

**Comment by kingjia90** [comment, marked as answer]
convert the resource to a `string` and use `imagecreatefromstring` along with `header`
```
        ...
        $avatar = $user->getImage();
        $imageString =  stream_get_contents($avatar);
        $image = imagecreatefromstring($imageString);
        header('Content-Type: image/png');
        imagepng($image);
```
or get the file path by
```
    $metaData = stream_get_meta_data($avatar);
    $filename = $metaData["uri"];
    echo $filename;
```

**Reply by idna001** [reply]
Thank you, the first approach works. I will figure out how to display this gd in twig. 
The second method is only possible if the backend is logged in. 

Greetings 
Andreas
